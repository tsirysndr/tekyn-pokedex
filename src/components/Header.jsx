import React from "react";
import PropTypes from "prop-types";
import { View } from "react-native";

export default function Header(props) {
  return <View></View>;
}
