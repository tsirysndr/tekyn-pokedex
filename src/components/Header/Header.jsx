import React from "react";
import PropTypes from "prop-types";
import { View } from "react-native";
import styled, { css } from "@emotion/native";
import DefaultText from "@Components/Text";
import { useFonts, Roboto_700Bold } from "@expo-google-fonts/roboto";
import BackIcon from "@Components/BackIcon";
import _ from "lodash";

const Text = styled(DefaultText)`
  font-size: 16px;
  color: #264653;
  flex: 1;
  text-align: center;
  margin-right: 50px;
  ${({ fontsLoaded }) =>
    fontsLoaded &&
    css`
      font-family: "Roboto_700Bold";
    `}
`;

const Container = styled.View`
  flex-direction: row;
  height: 50px;
  justify-content: center;
  align-items: center;
  ${({ backgroundColor }) =>
    backgroundColor &&
    css`
      background-color: ${backgroundColor};
    `}
`;

const Separator = styled.View`
  height: 1px;
  background-color: #f4a261;
`;

const BackButton = styled.TouchableOpacity`
  height: 50px;
  width: 50px;
  align-items: center;
  justify-content: center;
`;

export default function Header({ title, onGoBack, backgroundColor }) {
  const [fontsLoaded] = useFonts({ Roboto_700Bold });
  return (
    <>
      <Container backgroundColor={backgroundColor}>
        <BackButton onPress={onGoBack}>
          <BackIcon height={20} width={20} />
        </BackButton>
        <Text fontsLoaded>{_.capitalize(title)}</Text>
      </Container>
      <Separator />
    </>
  );
}

Header.defaultProps = {
  title: "",
  onGoBack: () => {},
};

Header.propTypes = {
  backgroundColor: PropTypes.string,
  title: PropTypes.string.isRequired,
  onGoBack: PropTypes.func,
};
