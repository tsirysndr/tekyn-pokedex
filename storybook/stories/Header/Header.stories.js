import React from "react";
import Header from "@Components/Header/Header";
import { storiesOf } from "@storybook/react-native";
import CenterView from "../CenterView";

storiesOf("Header", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("default", () => <Header></Header>);
