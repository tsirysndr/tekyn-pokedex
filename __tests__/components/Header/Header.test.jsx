import Header from "@Components/Header";
import { render } from "@testing-library/react-native";

test("renders the Header", () => {
  const tree = render(
    <Header />
  );
  expect(tree).toMatchSnapshot();
});
