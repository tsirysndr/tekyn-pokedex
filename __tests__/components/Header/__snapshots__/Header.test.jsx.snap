// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Header 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        fill="none"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={40}
        vbWidth={24}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M1.129 17.157c.203-.202.423-.373.655-.521L17.286 1.134a3.876 3.876 0 0 1 5.48 5.48L9.413 19.964l13.42 13.422a3.874 3.874 0 0 1-5.478 5.479L1.784 23.29a3.85 3.85 0 0 1-.655-.522A3.85 3.85 0 0 1 0 19.963a3.852 3.852 0 0 1 1.128-2.806Z"
            fill={4280698451}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      color="#fff"
      fontsLoaded={false}
      style={
        Array [
          Object {
            "color": "#fff",
          },
          Array [
            Object {
              "color": "#264653",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Roboto_700Bold",
              "fontSize": 16,
              "marginRight": 50,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    >
      
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f4a261",
          "height": 1,
        },
        undefined,
      ]
    }
  />,
]
`;
